cmd/frontend/internal/auth/userpasswd/reset_password.go

```diff
@@ -76,6 +78,61 @@ func HandleResetPasswordInit(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+var setPasswordEmailTemplates = txemail.MustValidate(txtypes.Templates{
+	Subject: `Set your Sourcegraph password`,
+	Text: `
+Your administrator created an account for you on Sourcegraph.
+
+To set the password for {{.Username}} on Sourcegraph, follow this link:
+
+  {{.URL}}
+`,
+	HTML: `
+<p>
+  Your administrator created an account for you on Sourcegraph.
+</p>
+
+<p><strong><a href="{{.URL}}">Set password for {{.Username}}</a></strong></p>
+`,
+})
+
+// Send the password reset email directly to the user on user creation by admin
+func HandleSetPasswordEmail(ctx context.Context, id int32) (string, error) {
+	e, _, err := db.UserEmails.GetPrimaryEmail(ctx, id)
+	if err != nil {
+		return "", errors.New("failed to lookup user email")
+	}
+
+	usr, err := db.Users.GetByID(ctx, id)
+	if err != nil {
+		return "", errors.New("failed to lookup user")
+	}
+
+	ru, err := backend.MakePasswordResetURL(ctx, id)
+	if err == db.ErrPasswordResetRateLimit {
+		return "", errors.New("too many password reset requests. try again in a few minutes")
+	} else if err != nil {
+		return "", errors.New("could not reset password")
```

Comment:
The actual error is swallowed:
```suggestion
		return "", errors.Wrap(err, "make password reset URL")
```